Packs/Druva/Integrations/DruvaEventCollector/DruvaEventCollector_test.py

```diff
@@ -0,0 +1,167 @@
+    """
+    Given:
+    - test module command
+
+    When:
+    - Pressing test button
+
+    Then:
+    - Test module failed
+    """
+    mocker.patch.object(mock_client, "_http_request", side_effect=DemistoException(message='Error: invalid_grant'))
+    with pytest.raises(DemistoException):
+        run_test_module(client=mock_client)
+
+
+EVENT_1 = {
+    "events": [
+        {
+            "eventID": 0,
+            "eventType": "Backup",
+            "profileName": "Default",
+            "inSyncUserName": "user name",
+            "clientVersion": "7.5.0r(23c42be5)",
+            "clientOS": "Office 365 Exchange Online",
+            "ip": "",
+            "inSyncUserEmail": "test@test.com",
+            "eventDetails": "",
+            "timestamp": "2024-05-25T18:52:48Z",
+            "inSyncUserID": 0,
+            "profileID": 0,
+            "initiator": None,
+            "inSyncDataSourceID": 0,
+            "eventState": "Backed up with Errors",
+            "inSyncDataSourceName": "Exchange Online",
+            "severity": 4,
+            "facility": 23
+        }
+    ],
+    "nextPageExists": False,
+    "tracker": "xxxx"
+}
+
+
+def test_get_events_command(mocker, mock_client):
+    """
+    Given:
+    - get_events command
+
+    When:
+    - running get events command
+
+    Then:
+    - events and tracker as expected
+    """
+    mocker.patch.object(mock_client, "search_events", return_value=EVENT_1)
+    events, tracker = get_events(client=mock_client)
+
+    assert tracker == 'xxxx'
+    assert len(events) == 1
+
+
+def test_refresh_access_token(mocker, mock_client):
+    """
+    Given:
+    - a mock client
+
+    When:
+    - running refresh_access_token method
+
+    Then:
+    - Ensure exception is thrown
+    - Ensure informative message was shown
+    """
+    client = Client(base_url="client_test")
+    mocker.patch.object(client, "_http_request", side_effect=DemistoException(message='Error: invalid_grant'))
+    with pytest.raises(DemistoException, match="Make sure Server URL, Client ID and Secret Key are correctly entered."):
+        client.refresh_access_token(encoded_credentials=b"test")
+
+
+EVENT_2 = {
+    "events": [
+        {
+            "eventID": 1,
+            "eventType": "Backup",
+            "profileName": "Default",
+            "inSyncUserName": "user name",
+            "clientVersion": "7.5.0r(23c42be5)",
+            "clientOS": "Office 365 Exchange Online",
+            "ip": "",
+            "inSyncUserEmail": "test@test.com",
+            "eventDetails": "",
+            "timestamp": "2024-05-25T18:52:48Z",
+            "inSyncUserID": 1,
+            "profileID": 1,
+            "initiator": None,
+            "inSyncDataSourceID": 1,
+            "eventState": "Backed up with Errors",
+            "inSyncDataSourceName": "Exchange Online",
+            "severity": 4,
+            "facility": 23
+        }
+    ],
+    "nextPageExists": False,
+    "tracker": "yyyy"
```

Comment:
Can there be a response where there's no next page, but there _is_ a tracker? 🤔

Reply:
yes, it points to the end of the events